src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10937,9 +10937,6 @@ private ExpressionSyntax ParseSubExpression(Precedence precedence)
 
         private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
         {
-            ExpressionSyntax leftOperand;
-            Precedence newPrecedence = 0;
```

Comment:
an example of very confusing local state.  inner blocks would set this, to use only within that block.  But the value was never read afterwards.  Removed entirely as having multiple precedence values to have to understand in a precedence parser makes things much harder to reason about :)